### ISSUE TITLE
poor performance on conv2d with ow=1

### ISSUE THREAD

**Issue by lxn179208** [tensorflow/tensorflow#23514]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):linux
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:no
- TensorFlow installed from (source or binary):binary
- TensorFlow version (use command below):'v1.11.0-0-gc19e293', '1.11.0'
- Python version:2.7.5
- Bazel version (if compiling from source):no
- GCC/Compiler version (if compiling from source):no
- CUDA/cuDNN version:no
- GPU model and memory:no

I try to optimize the voice synthesize model, with a lot of conv1d invoke fllowing the steps of https://www.tensorflow.org/guide/performance/overview#optimizing_for_cpu, I found the results on tensorflow v1.11.0 with mkl version is slower 50% than the same tensorflow without mkl.

First of all, based on https://github.com/intel/mkl-dnn/issues/285,  I test conv2d NCHW with respectively oh=1 or ow=1. I found ow=1 slower slower than oh=1.

So I doubt how to optimize cpu performance on conv1d ???

-----------------------------------------------------------------------------

### Environment
Intel MKL-DNN includes hardware-specific optimizations and may behave
differently on depending on the compiler and build environment. Include
the following information to help reproduce the issue:
* Intel(R) Xeon(R) CPU E5-2682 v4 @ 2.50GHz
* company internal version based on centos
* 4.9.2
* empty value
* no cmake
* tensorflow 1.11

### Steps to reproduce

```
isntall fllowing https://software.intel.com/en-us/articles/intel-optimization-for-tensorflow-installation-guide
# Python 2.7
pip install https://storage.googleapis.com/intel-optimized-tensorflow/tensorflow-1.11.0-cp27-cp27mu-linux_x86_64.whl 

```
ow =1  cmd

```
MKLDNN_VERBOSE=1 KMP_BLOCKTIME=0 CUDA_VISIBLE_DEVICES=-1  OMP_NUM_THREADS=32 python conv2d.py  ow

ow=1
2018-11-05 15:42:35.614687: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
start
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,32x128x102400x1,85.6531
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_hwio out:f32_OIhw8i8o,num:1,128x128x3x1,0.134766
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:undef fdst:nChw8c,alg:convolution_direct,mb32_g1ic128oc128_ih102400oh102398kh3sh1dh0ph0_iw1ow1kw1sw1dw0pw0,991.858
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,32x128x102398x1,174.348
data_format_2d=NCHW 100 epoch cost 2.131751
```
oh =1  cmd

```
MKLDNN_VERBOSE=1 KMP_BLOCKTIME=0 CUDA_VISIBLE_DEVICES=-1  OMP_NUM_THREADS=32 python conv2d.py  oh
oh=1
2018-11-05 15:42:05.224622: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
start
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nchw out:f32_nChw8c,num:1,32x128x1x102400,91.1848
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_hwio out:f32_OIhw8i8o,num:1,128x128x1x3,0.152832
mkldnn_verbose,exec,convolution,jit:avx2,forward_training,fsrc:nChw8c fwei:OIhw8i8o fbia:undef fdst:nChw8c,alg:convolution_direct,mb32_g1ic128oc128_ih1oh1kh1sh1dh0ph0_iw102400ow102398kw3sw1dw0pw0,835.947
mkldnn_verbose,exec,reorder,jit:uni,undef,in:f32_nChw8c out:f32_nchw,num:1,32x128x1x102398,199.113
data_format_2d=NCHW 100 epoch cost 1.990710

```

### the code shows

```
import os
import sys
import tensorflow as tf
import numpy as np
import time

"""
default NWC
input=[batch, in_width, in_channels]  filter=[ filter_width, in_channels, out_channels]
=>tf.nn.conv2d  [batch,1,in_width, in_channels]  [1, filter_width, in_channels, out_channels]

mkl NCW
input=[batch, in_channels, in_width] filter=[filter_width, in_channels, out_channels]
=>tf.nn.conv2d []
"""

batch = 32
in_width = 102400
in_height = 1
in_channels = 128
filter_width = 3
filter_height = 1
out_channels = 128
data_format_2d = 'NCHW'
n = 100
start = 99
input_2d = {}

if len(sys.argv) == 2 and sys.argv[1] == "ow":
    print("ow=1")
    in_width,in_height = in_height, in_width
    filter_width,filter_height = filter_height, filter_width
else:
    print("oh=1")
if data_format_2d == "NCHW":
    input_ = np.array(np.arange(1, 1 + batch*in_width*in_height*in_channels).reshape([batch, in_channels, in_height,in_width]), dtype=np.float32)
else:
    input_ = np.array(np.arange(1, 1 + batch*in_width*in_height*in_channels).reshape([batch, in_height,in_width,in_channels]), dtype=np.float32)
for i in range(start,n):
    input_2d[i] = input_ + i*n*i*1.0
kernel_2d = np.array(np.arange(1, 1 + filter_width*filter_height*in_channels*out_channels), dtype=np.float32).reshape([filter_height, filter_width, in_channels, out_channels])
a = tf.placeholder(dtype=tf.float32)
conv2d = tf.nn.conv2d(a, kernel_2d, [1,1,1,1], 'VALID', data_format=data_format_2d)

config = tf.ConfigProto()
config.intra_op_parallelism_threads = 32
config.inter_op_parallelism_threads = 2
sess = tf.Session(config=config)

print("start")
sess.run(tf.global_variables_initializer())
start_ts = time.time()
for i in range(start,n):
    sess.run(conv2d, feed_dict={a:input_2d[i]})
print("data_format_2d=%s %d epoch cost %f" % (data_format_2d,n, (time.time()) - start_ts))
```

**Comment by tensorflowbutler** [comment]
Nagging Assignee @TensorFlow-MKL: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by jingxu10** [comment]
Hi,
Regarding to the performance gap of Intel TensorFlow, please check my reply in https://github.com/intel/mkl-dnn/issues/285.

Since the calculation is 1D convolution, why not trying conv1d like the following snippet?
```
import tensorflow as tf
import numpy as np
import time

"""
default NWC
input=[batch, in_width, in_channels]  filter=[ filter_width, in_channels, out_channels]
=>tf.nn.conv2d  [batch,1,in_width, in_channels]  [1, filter_width, in_channels, out_channels]

mkl NCW
input=[batch, in_channels, in_width] filter=[filter_width, in_channels, out_channels]
=>tf.nn.conv2d []
"""

batch = 32
in_width = 102400
in_channels = 128
filter_width = 3
out_channels = 128
data_format= "NCW"
n = 10

if data_format == "NCW":
    input_ = np.random.rand(batch, in_channels, in_width)
# print(input_)
kernel_1d = np.random.rand(filter_width, in_channels, out_channels).astype(np.float32)
a = tf.placeholder(dtype=tf.float32)
conv1d = tf.nn.conv1d(a, kernel_1d, 1, 'VALID', data_format=data_format)

config = tf.ConfigProto()
# config.intra_op_parallelism_threads = 32
# config.inter_op_parallelism_threads = 2
sess = tf.Session(config=config)

print("start")
sess.run(tf.global_variables_initializer())
for i in range(0, 10):
    # print("Warm-up: %d" % (i))
    sess.run(conv1d, feed_dict={a:input_})
start_ts = time.time()
for i in range(0, n):
    # print("Test: %d" % (i))
    sess.run(conv1d, feed_dict={a:input_})
print("data_format_1d=%s %d epoch cost %f" % (data_format, n, (time.time()) - start_ts))
print("shape:")
print(np.shape(input_))
```

**Comment by TensorFlow-MKL** [comment]
closing due to inactivity.
